src/noteeditorlogic.cpp

```diff
@@ -162,13 +194,202 @@ void NoteEditorLogic::onTextEditTextChanged()
+        || startLinePosition > endLinePosition) {
+        qWarning() << "Invalid positions provided.";
+        return inputText;
+    }
+
+    QStringList textToMove = lines.mid(startLinePosition, endLinePosition - startLinePosition + 1);
+    lines.erase(lines.begin() + startLinePosition, lines.begin() + endLinePosition + 1);
+
+    if (isColumns && startLinePosition > originalNewLinePosition
+        && lines[startLinePosition].isEmpty()) {
+        // moving column backward
+        lines.remove(startLinePosition);
+    }
+
+    if (newLinePosition > startLinePosition)
+        newLinePosition -= textToMove.size();
+
+    for (int i = 0; i < textToMove.size(); ++i)
+        lines.insert(newLinePosition + i, textToMove.at(i));
+
+    if (isColumns) {
+        if (startLinePosition < originalNewLinePosition) {
+            // moving column forward
+            if (lines[startLinePosition].isEmpty())
+                lines.removeAt(startLinePosition);
+            lines.insert(newLinePosition - 1, "");
+        } else {
+            // moving column backward
+            lines.insert(newLinePosition + textToMove.size(), "");
+        }
+    }
+
+    return lines.join("\n");
+}
+
+// TODO: Rearrange tasks or colums witout clearing and setting the entire text, rather just change
+// what's needed
+void NoteEditorLogic::rearrangeTasksInTextEditor(int startLinePosition, int endLinePosition,
+                                                 int newLinePosition)
+{
+    //    !if (!kanbanReadOnly) {
+    if (startLinePosition != newLinePosition) {
+        QString newText = moveTextToNewLinePosition(m_textEdit->toPlainText(), startLinePosition,
+                                                    endLinePosition, newLinePosition);
+        m_textEdit->setText(newText);
+    }
+}
+
+void NoteEditorLogic::rearrangeColumnsInTextEditor(int startLinePosition, int endLinePosition,
+                                                   int newLinePosition)
+{
+    //    !if (!kanbanReadOnly) {
+    if (startLinePosition != newLinePosition) {
+        QString newText = moveTextToNewLinePosition(m_textEdit->toPlainText(), startLinePosition,
+                                                    endLinePosition, newLinePosition, true);
+        m_textEdit->setText(newText);
+    }
+}
+
+// QJsonArray([
+// {
+//    "title":"TODO",
+//    "tasks":[
+//              {"checked":false,"text":"todo 1"},
+//              {"checked":false,"text":"todo 2"}]
+// },
+// {"
+//      "title":"In Progress"}
+//       tasks":[
+//              {"checked":false,"text":"in progress 1"}],
+// }
+// ])
+void NoteEditorLogic::checkForTasksInEditor()
+{
+    QStringList lines = m_textEdit->toPlainText().split("\n");
+    QJsonArray data;
+    QJsonObject currentColumn;
+    QJsonArray tasks;
+    QString currentTitle = "Untitled";
+    QRegularExpression taskRegex("^\\s*-\\s*\\[(x| )\\]\\s*(.*)$");
```

Comment:
This doesn't work for something like this:

```
* [ ] TODO 4
```

But instead of using regex, maybe there's a better way to check whether the line is a list item?

Reply:
Right, maybe regex is overkill here. That was ChatGPT's idea hah. Maybe a simple `find('- [ ]") / '- [x]")` should do?

Ah I didn't know that's another way people use that. It should be easy to support it.

Reply:
Sure, but I think it would be more robust if this logic is tied together with the markdown highlighter. The user gets immediate feedback to know if it's a list item (because the color changes) and therefore knows for sure that it will show up in the kanban view. 

If something goes wrong with the regex for any reason (like the above bug), the user has to guess why it went wrong.

Reply:
Gotcha. Yes, sounds better.

Reply:
Now supporting other types of task expressions and dropping regex via https://github.com/nuttyartist/notes/pull/574/commits/406c6d2c47128203a78420899acf5d43fa8b4463